tools/cache.py

```diff
@@ -36,6 +36,8 @@ def __init__(self, dirname, use_subdir=True):
         subdir += '-lto'
       if shared.Settings.RELOCATABLE:
         subdir += '-pic'
+      if shared.Settings.MEMORY64:
+        subdir += '-memory64'
```

Comment:
`-wasm64`?   (to match llvm's name for this)

Reply:
Will we have a `table64` some day? If so what would LLVM call the mode with `memory64` but not `table64`, and vice versa?

Reply:
the feature on the Wasm side is called memory64 (and uses this name thru-out wabt, binaryen, and this PR so far.